framework/ui/src/main/java/com/google/android/catalog/framework/ui/CatalogScreen.kt

```diff
@@ -104,37 +118,91 @@ internal fun CatalogScreen(
-                        searchState = false
-                    },
-                    onValueChange = {
-                        searchTerm = it
+            AnimatedContent(
+                targetState = searchState,
+                transitionSpec = {
+                    if (targetState) {
+                        expandHorizontally { it / 8 } + fadeIn(tween(100)) with fadeOut()
+                    } else {
+                        fadeIn(tween(500, delayMillis = 100)) with
+                            shrinkHorizontally(tween(400)) { it / 8 } + fadeOut(tween(400))
                     }
-                )
-            } else {
-                CatalogTopAppBar(onSearch = { searchState = true })
+                },
+                contentAlignment = Alignment.CenterEnd
+            ) {
+                if (it) {
+                    SearchTopAppBar(
+                        searchTerm = searchTerm,
+                        focusRequester = focusRequester,
+                        onSearch = {
+                            searchState = false
+                        },
+                        onClear = {
+                            searchState = false
+                        },
+                        onValueChange = {
+                            searchTerm = it
+                        }
+                    )
+                } else {
+                    CatalogTopAppBar(
+                        isDualPane = isExpandedScreen,
+                        selectedSample = selectedSample,
+                        onSearch = { searchState = true },
+                        onExpand = { launchSample(selectedSample!!) }
+                    )
+                }
             }
         }
     ) { paddingValues ->
-        SamplesList(
-            paddingValues,
-            filters,
-            selectedFilters,
-            displayedSamples,
-            catalogSettings.cardAppearance
-        ) {
-            launchSample(it)
-            searchState = false
-        }
+        AdaptivePane(
+            paddingValues = paddingValues,
+            start = {
+                SamplesList(
+                    paddingValues,
+                    filters,
+                    selectedFilters,
+                    displayedSamples,
+                    selectedSample,
+                    catalogSettings.cardAppearance
+                ) {
+                    if (!isExpandedScreen || it.target !is CatalogTarget.TargetComposable) {
+                        launchSample(it)
+                    }
+                    if (it.target is CatalogTarget.TargetComposable) {
+                        selectedSample = it
+                    }
+                    searchState = false
+                }
+            },
+            end = {
+                val content =
+                    (selectedSample?.target as? CatalogTarget.TargetComposable)?.composable
+                if (content != null) {
+                    content.invoke()
```

Comment:
Related to the state preservation: this content will be invoked in a different place here, and if it is launched separately. So if there was a text field in the sample, and then you launched the sample directly, I'd expect the text field state would be reset. That's probably fine given the constraints of the samples, but just wanted to highlight it.

Reply:
Do you mean that when it switches from dual pane to full screen it will lose state since it's launch in a different place?

If so, I guess that's okay for samples. In order to solve that I would need to skip the jetpack navigation and do the magic directly in this composable right?

Reply:
Exactly, this line would need to have the same state as the line here:
https://github.com/google/casa-android/blob/aba16135be7f1378dcac8209ac046389a0400bab/framework/ui/src/main/java/com/google/android/catalog/framework/ui/CatalogNavigation.kt#L77

or somehow hoist a `movableContentOf` to be shared between the two.

For this case that's probably fine, but would lead to a frustrating experience for a general use case in an app along the lines of drafting a form in the detail pane (with state stored inside it), and then if you open to fullscreen it wipes out that state.

Reply:
I see. I will look into it if I see that it causes issues with the samples, but I think for now it's fine. Thanks for highlighting that